(DOCSP-36281): Add new appservices deploy CLI docs (#669)
mongodb/docs-app-services: source/apps/deploy-github.txt

Enable Automatic Deployment with GitHub
---------------------------------------

You can enable automatic deployment with GitHub using the App Services UI,
the {+cli+}, or the App Services Admin API.

.. tabs-realm-admin-interfaces::
   
   .. tab::
      :tabid: ui

      .. procedure::

         .. step:: Install the Atlas App Services GitHub App

            In order to automatically deploy based on a GitHub repository,
            Atlas App Services requires that you install a GitHub app that has, at minimum,
            read access to the repository.

            To install the app, click :guilabel:`Deployment` in the left navigation menu of the App Services UI.
            Select the :guilabel:`Configuration` tab and then click :guilabel:`Install
            App Services on GitHub`. A new browser window will open to the GitHub application
            installation flow.

            .. note:: GitHub Authentication

               GitHub may require you to provide your GitHub account credentials before
               you can install the app.

            Select the GitHub account or organization for which you want to install the
            app:

>             .. figure:: /images/github-app-install-location.png

Author: Sounds like a good discussion to have as a team! I'll resolve this comment here, and make a note for us to revisit process around images in procedures.

               You can see the currently deployed version of your
               application as well as a historical log of previous
               deployments in the :ref:`Deployment History
               <deployment-history>` table in the App Services UI.

   .. tab::
      :tabid: cli

      .. versionadded:: 1.1.0

      You can use the CLI to configure, enable, or disable automatic deployment.

      .. procedure::
      
         .. step:: Authenticate a MongoDB Atlas User

            .. include:: /includes/cli-login.rst

         .. step:: Run the Deploy Configure Command

            The {+cli+} can walk you through the process of configuring automatic
            deployment. This is an interactive command.

            .. code-block:: shell

               {+cli-bin+} deploy configure

            The command also supports additional flags that you can optionally 
            include. For information on these flags, refer to 
            :ref:`appservices-deploy-configure`.

            {+cli+} immediately enables automatic deployment when you complete
            this command. You don't need to run the ``enable`` command as
            an additional step.

>             .. tip:: This functionality relies on a GitHub beta feature

Reviewer: ```suggestion
            .. tip:: Configuring automatic deployment via CLI relies on a GitHub beta feature
```

               The ability to configure automatic deployment using {+cli+}
               relies on a GitHub feature that is currently in beta. If 
               GitHub makes breaking changes to this feature, you may be unable
               to configure automatic deployment using the {+cli+}. In this
               case, use the App Services UI or Admin API to configure 
               automatic deployment.

>                This only applies to configuring automatic deployment. If you 

Reviewer: would it be accurate to say that this only applies to the ``appservices deploy configure`` command, and the ``deploy enable`` and ``deploy disable`` commands don't rely on the feature?

Author: Updating the verbiage to clarify details!

               have previously configured it for your application, the
               functionality to enable or disable it using the {+cli+}
               does not rely on this beta GitHub feature.

         .. step:: Disable, Enable, or View Configuration Details (Optional)

            You can use the {+cli+} to view configuration details after you have
            configured automatic deployment.

            .. code-block:: shell

               {+cli-bin+} deploy describe

>             You can disable and enable automatic deployment using the {+cli+}. 

Reviewer: ```suggestion
            You can also disable and enable automatic deployment using the {+cli+}. 
```

            For more information about these commands and their options, refer 
            to :ref:`appservices-deploy`.

   .. tab::
      :tabid: api

      .. procedure::

         .. step:: Authenticate a MongoDB Atlas User

            .. include:: /includes/api-authenticate-instructions.rst

         .. step:: Get an Existing Deployment Configuration (Optional)

            You can get the existing deployment configuration for your App by 
            making a ``GET`` request to the :admin-api-endpoint:`Get the 
            Deployment Configuration <operation/adminGetDeploymentConfig>`
            endpoint. 

            .. code-block:: bash

               curl -X GET \
                  https://services.cloud.mongodb.com/api/admin/v3.0/groups/{groupId}/apps/{appId}/deploy/config \
                  -H 'Authorization: Bearer <access_token>'

         .. step:: Update the Deployment Configuration

>             You can get the existing deployment configuration for your App by 

Reviewer: ```suggestion
            You can update the existing deployment configuration for your App by 
```

.. _deploy-github-make-changes-cli:

Make Changes from the CLI
-------------------------

> Previously, we have recommended avoiding avoid making changes to your app 

Reviewer: ```suggestion
Previously, we have recommended that you avoid making changes to your app 
```

with {+cli+} after configuring automatic deployment. Starting with 

> {+cli+} v1.1.0 and newer, you can safely use the {+cli+} to make changes to 

Reviewer: had to double check but this is what we agreed on to use
```suggestion
{+cli+} v1.1.0 and later, you can safely use the {+cli+} to make changes to 
```

your app.

> **{+cli+} version 1.1.0 and newer**

Reviewer: ```suggestion
**{+cli+} version 1.1.0 and later**
```

> In {+cli+} v1.1.0 and newer, a push from the {+cli+} automatically generates

Reviewer: ```suggestion
In {+cli+} v1.1.0 and earlier, a push from the {+cli+} automatically generates
```

a corresponding commit in GitHub. This keeps your code changes synchronized
with your version-controlled GitHub source of truth.

> **{+cli+} version 1.0.3 and older**

Reviewer: ```suggestion
**{+cli+} version 1.0.3 and earlier**
```

> In {+cli+} v1.0.3 and older, avoid making changes to your app with the {+cli+}

Reviewer: ```suggestion
In {+cli+} v1.0.3 and earlier, avoid making changes to your app with the {+cli+}
```

.. _deploy-github-make-changes-cli:

Make Changes from the CLI
-------------------------

Previously, we have recommended that you avoid making changes to your app 
with {+cli+} after configuring automatic deployment. Starting with 
{+cli+} v1.1.0 and later, you can safely use the {+cli+} to make changes to 
your app.

**{+cli+} version 1.1.0 and later**

> In {+cli+} v1.1.0 and earlier, a push from the {+cli+} automatically generates

Reviewer 2: ```suggestion
In {+cli+} v1.1.0 and later, a push from the {+cli+} automatically generates
```

  GitHub Deployment, those changes are automatically committed back to your 
  linked repository.

- If you would like to use {+cli+} to update configuration files after enabling 

>   Automatic GitHub Deployment, use {+cli+} v1.1.0 or newer.

Reviewer: ```suggestion
  Automatic GitHub Deployment, use {+cli+} v1.1.0 or later.
```
